stellarphot/settings/models.py

```diff
@@ -80,3 +91,107 @@ class PhotometryFileSettings(BaseModel):
     aperture_locations_file: Path = Field(
         filter_pattern=["*.ecsv", "*.csv"], default=""
     )
+
+
+class TimeType(Time):
+    @classmethod
+    def __get_validators__(cls):
+        yield cls.validate
+
+    @classmethod
+    def validate(cls, v):
+        return Time(v)
+
+
+class SkyCoordType(SkyCoord):
+    @classmethod
+    def __get_validators__(cls):
+        yield cls.validate
+
+    @classmethod
+    def validate(cls, v):
+        return SkyCoord(v)
+
+
+class Exoplanet(BaseModel):
+    """
+    Create an object representing an Exoplanet.
+
+    Parameters
+    ----------
+
+    epoch : `astropy.time.Time`, optional
+        Epoch of the exoplanet.
+
+    period : `astropy.units.Quantity`, optional
+        Period of the exoplanet.
+
+    Identifier : str
+        Identifier of the exoplanet.
+
+    coordinate : `astropy.coordinates.SkyCoord`
+        Coordinates of the exoplanet.
+
+    depth : float
+        Depth of the exoplanet.
+
+    duration : `astropy.units.Quantity`, optional
+        Duration of the exoplanet transit.
+
+    Examples
+    --------
+
+    To create an `Exoplanet` object, you can pass in the epoch,
+     period, identifier, coordinate, depth, and duration as keyword arguments:
+
+    >>> planet  = Exoplanet(epoch=Time(2455909.29280, format="jd"),
+    ...                     period=1.21749 * u.day,
+    ...                     identifier="KELT-1b",
+    ...                     coordinate=SkyCoord(ra="00:01:26.9169",
+    ...                                         dec="+39:23:01.7821",
+    ...                                         frame="icrs",
+    ...                                         unit=("hour", "degree")),
+    ...                     depth=0.006,
+    ...                     duration=120 * u.min)
+    """
+
+    epoch: TimeType | None = None
+    period: QuantityType | None = None
+    identifier: str
+    coordinate: SkyCoordType
+    depth: float | None = None
+    duration: QuantityType | None = None
+
+    class Config:
+        validate_all = True
+        validate_assignment = True
+        extra = "forbid"
+        json_encoders = {
+            Quantity: lambda v: f"{v.value} {v.unit}",
+            QuantityType: lambda v: f"{v.value} {v.unit}",
+            Time: lambda v: f"{v.value}",
+        }
+
+    @classmethod
```

Comment:
this needs one more decorator to finish turning it into something that pydantic actually validates, something like `@validator("period")` (after importing `validator` from `pydantic`).

See https://github.com/feder-observatory/stellarphot/blob/main/stellarphot/core.py#L289 for an example. That is why leaving the units off didn't generate a validation error.  Let's ,make sure it does by only making those two values bad.

Reply:
@Tanner728 -- do you have time to add this change or should I wrap it up instead?

Reply:
I should have time tonight to do it.

Reply:
Looks like you left out the "period" argument to `@validator`....